test/core/security/rbac_translator_test.cc

```diff
@@ -53,9 +68,39 @@ MATCHER_P4(EqualsHeader, expected_name, expected_matcher_type,
              : arg->header_matcher.string_matcher() == expected_matcher_value;
 }
 
+class TestAuditLoggerFactory : public AuditLoggerFactory {
+ public:
+  class TestAuditLoggerConfig : public AuditLoggerFactory::Config {
+    absl::string_view name() const override { return kLoggerName; }
+    std::string ToString() const override { return ""; }
+  };
+  absl::string_view name() const override { return kLoggerName; }
+  absl::StatusOr<std::unique_ptr<AuditLoggerFactory::Config>>
+  ParseAuditLoggerConfig(const Json& json) override {
+    if (json.object().find("bad") != json.object().end()) {
+      return absl::InvalidArgumentError("bad logger config.");
+    }
+    return std::make_unique<TestAuditLoggerConfig>();
+  }
+  std::unique_ptr<AuditLogger> CreateAuditLogger(
+      std::unique_ptr<AuditLoggerFactory::Config>) override {
+    Crash("unreachable");
```

Comment:
Can you add a comment explaining this, it's not immediately clear to me why `CreateAuditLogger` should Crash?

Reply:
`rbac_translator.cc` shouldn't call this method. So having it crash in this test confirms this fact.

Reply:
Ah so it's just a method that has to be there, but since it shouldn't be called we crash to ensure it isn't called